test/integration/authorization_test.go

```
@@ -609,55 +612,205 @@ type subjectAccessReviewTest struct {
-				failMessage = fmt.Sprintf("%s: expected\n\t%v\ngot\n\t%v", test.description, test.err, err)
-				return false, nil
+	{
+		failMessage := ""
+		err := wait.Poll(testutil.PolicyCachePollInterval, testutil.PolicyCachePollTimeout, func() (bool, error) {
+			var err error
+			var actualResponse *authorizationapi.SubjectAccessReviewResponse
+			if test.localReview != nil {
+				actualResponse, err = test.localInterface.Create(test.localReview)
+			} else {
+				actualResponse, err = test.clusterInterface.Create(test.clusterReview)
 			}
-		} else {
-			if err != nil {
-				failMessage = fmt.Sprintf("%s: unexpected error: %v", test.description, err)
+			if len(test.err) > 0 {
+				if err == nil {
+					failMessage = fmt.Sprintf("%s: Expected error: %v", test.description, test.err)
+					return false, nil
+				} else if !strings.HasPrefix(err.Error(), test.err) {
+					failMessage = fmt.Sprintf("%s: expected\n\t%v\ngot\n\t%v", test.description, test.err, err)
+					return false, nil
+				}
+			} else {
+				if err != nil {
+					failMessage = fmt.Sprintf("%s: unexpected error: %v", test.description, err)
+					return false, nil
+				}
+			}
+
+			if (actualResponse.Namespace != test.response.Namespace) ||
+				(actualResponse.Allowed != test.response.Allowed) ||
+				(!strings.HasPrefix(actualResponse.Reason, test.response.Reason)) {
+				if test.localReview != nil {
+					failMessage = fmt.Sprintf("%s: from local review\n\t%#v\nexpected\n\t%#v\ngot\n\t%#v", test.description, test.localReview, &test.response, actualResponse)
+				} else {
+					failMessage = fmt.Sprintf("%s: from review\n\t%#v\nexpected\n\t%#v\ngot\n\t%#v", test.description, test.clusterReview, &test.response, actualResponse)
+				}
 				return false, nil
 			}
+
+			failMessage = ""
+			return true, nil
+		})
+
+		if err != nil {
+			t.Error(err)
+		}
+		if len(failMessage) != 0 {
+			t.Error(failMessage)
+		}
+	}
+
+	if test.kubeAuthInterface != nil {
```

Why would this ever be nil?  Seems like I'd always want to do this check and rather than accidentally skip it, we should fail.